### ISSUE TITLE
Only my mic gets connected to a Lync meeting

### ISSUE THREAD

**Issue by brianjmurrell** [tieto/sipe#163]
I've built c817c0c6 on Fedora 26 and having difficulties with it.

It takes a long time to connect to a call with it sitting in "Calling..."  for about 30 seconds before the call transitions to "Call in progress"

But more importantly, when I get connected, pidgin opens a "recording" stream from PulseAudio and sends my voice/mic to the meeting but it never opens up a "playback" stream to send the meeting's audio to my headset.

**Comment by brianjmurrell** [comment]
No ideas here?  This unfortunately makes pidgin-sipe useless for me since I can't talk to anyone (well, I can talk, just cannot hear) or join meetings.

This coincided with a Fedora 25->26 upgrade if that's at all useful in providing any ideas.  Maybe I have some other component that is not working?

Happy to provide whatever additional information might be needed.

**Comment by brianjmurrell** [comment]
A lightbulb just went on above my head and I looked into libnice on Fedora 26.  It's a heavily patched 0.1.13.  :-(  I just confirmed updating to stock upstream libnice 0.1.14 resolves this issue.

But I have also taken notice of https://github.com/tieto/libnice/compare/0.1.14...launchpad.  Are any of those seriously worth having?  Any attempts to get them upstream?

**Comment by caio2k** [comment]
Thanks for the tip, it solved my audio issues in Debian Stretch. Upgrading libnice from 0.1.13 to 0.1.14 (from Debian Buster) + recompiling pidgin-sipe package fixed audio calling.

**Comment by dwmw2** [comment]
@brianjmurrell Yeah, libnice in Fedora of that vintage was 0.1.13 with everything that we needed at the time to make SIPE work nicely. Now that 0.1.14 is actually released, I see that the Fedora packages are updated. Although I think there are more post-0.1.14 changes which would also be useful now...

**Comment by brianjmurrell** [comment]
@dwmw2 Yeah.  Even with libnice 0.1.14 (from Fedora) I still get frequent one-way audio (opposite of what I was reporting here -- I can hear but not be heard).  But it's frequency is intermittent.  In one call I will experience it and another I won't.  I can go in long spurts of calls with one-way audio, then one or two work and then a string where I get one-way audio again.

I can even call the "Test Call" facility and confirm that my audio can be heard right before a call and then not have my audio work in the call .

It's all really quite unstable, unfortunately and I have taken to restarting Pidgin after every call now.  It's just far too unstable after a call and the odds that the next call will be problematic drop somewhat if I restart after each call.

Even with those measures, I probably end up dialing (from a land-line) into more calls than I am able to participate in with Pidgin due to the flakiness of it.

**Comment by dwmw2** [comment]
Could that be the same as https://github.com/awslabs/pidgin-chime/issues/20 ?

**Comment by dwmw2** [comment]
And speaking of restarting Pidgin, it would also be interesting to know if you recognise https://github.com/awslabs/pidgin-chime/issues/21

**Comment by brianjmurrell** [comment]
> Could that be the same as awslabs/pidgin-chime#20 ?

I don't think so.  When others cannot hear me, I don't see any sign of muting in Pidgin or PulseAudio.

> And speaking of restarting Pidgin, it would also be interesting to know if you recognise awslabs/pidgin-chime#21

That one sounds like this particular issue but TBH, I don't actually experience this particular issue very often.  Not nearly (by very far) as much as others not being able to hear me.  I can't remember the last time it was the other way around, where others heard me but I didn't hear others.

**Comment by dwmw2** [comment]
The muting is protocol-specific. Chime shows us participants' volume in real time, and if our own outbound frames have no audio attached (they are sent from a timer when we have no audio flowing) we appear "muted". In your case it would simply be an absence of outbound audio.

**Comment by brianjmurrell** [comment]
> In your case it would simply be an absence of outbound audio.

So that particular symptom is very common in my experience.  And that's a GStreamer race you say?  I'm not sure that rejoining when it's Lync fixes the problem.  I will pay more attention to that in the future when I experience this.
